sklearn/tests/test_cross_validation.py

```diff
@@ -160,28 +161,39 @@ def test_kfold_valueerrors():
 
     # Check that a warning is raised if the least populated class has too few
     # members.
-    y = [3, 3, -1, -1, 2]
+    y1 = [3, 3, -1]
 
     cv = assert_warns_message(Warning, "The least populated class",
-                              cval.StratifiedKFold, y, 3)
+                              cval.StratifiedKFold, y1, 2)
 
     # Check that despite the warning the folds are still computed even
     # though all the classes are not necessarily represented at on each
     # side of the split at each split
-    check_cv_coverage(cv, expected_n_iter=3, n_samples=len(y))
+    check_cv_coverage(cv, expected_n_iter=2, n_samples=len(y1))
+
+    # Check that errors are raised if all n_labels for individual
+    # classes are less than n_folds.
+    y2 = [3, 3, -1, -1, 2]
+
+    assert_raises(ValueError, cval.StratifiedKFold, y2, 3)
 
     # Error when number of folds is <= 1
     assert_raises(ValueError, cval.KFold, 2, 0)
     assert_raises(ValueError, cval.KFold, 2, 1)
-    assert_raises(ValueError, cval.StratifiedKFold, y, 0)
-    assert_raises(ValueError, cval.StratifiedKFold, y, 1)
+    # String to check if ValueError is not from a different error
+    error_string = ("k-fold cross validation requires at least one"
+                    " train / test split")
+    assert_raise_message(ValueError, error_string,
+                         cval.StratifiedKFold, y2, 0)
+    assert_raise_message(ValueError, error_string,
```

Comment:
can you change this back to y? In general, we don't modify previous tests unless absolutely required.